Add HomeLink docs (#37098)
home-assistant/home-assistant.github.io: source/_integrations/gentex_homelink.markdown

---
title: HomeLink
description: Instructions on how to set up your HomeLink devices with Home Assistant.
ha_category:
  - Binary sensor

> ha_release: 2025.4

Reviewer: ```suggestion
ha_release: 2026.1
```

---
title: HomeLink

> description: Instructions on how to setup your HomeLink enabled car with Home Assistant.

Reviewer: Is this only about integrating cars via HomeLink? Or does it apply to other devices?

Author: HomeLink is an [in-vehicle product](https://www.gentex.com/products-technology/automotive/connected-car), but it's fine for us to make to just call them "devices". Some are [rear-view mirrors](https://data.voxxelectronics.com/gentex/frameless-mirrors-homelink/), some are [embedded hardware modules](https://shop.tesla.com/product/model-3y-automatic-garage-opener).

ha_category:
  - Binary sensor
ha_release: 2025.3
ha_iot_class: Cloud Push
ha_config_flow: true
ha_codeowners:
  - '@niaexa'
  - '@ryanjones-gentex'
ha_domain: gentex_homelink
ha_platforms:
  - binary_sensor
ha_integration_type: integration
ha_quality_scale: bronze
---

> The HomeLink app allows [HomeLink](https://homelink.com) owners to trigger smart home routines in a variety of cloud-based home automation platforms from the comfort of your vehicle. When you arrive home, one HomeLink button press can adjust your thermostat, turn on your lights, disarm the security system, unlock the door, and begin playing your favorite music.

Reviewer: ```suggestion
The **HomeLink** {% term integration %} allows you to integrate [HomeLink](https://homelink.com) devices in Home Assistant. The HomeLink app allows [HomeLink](https://homelink.com) owners to trigger smart home routines in a variety of cloud-based home automation platforms from the comfort of their vehicle. When you arrive home, one HomeLink button press can adjust your thermostat, turn on your lights, disarm the security system, unlock the door, and begin playing your favorite music.
```
The first sentence should explain what the integration does. I provided an example, but please feel free to rephrase.

Author: Thanks for the suggestions, the overview has been updated.

Download the HomeLink app by scanning this QR code or following links to our app store pages:

![HomeLink app QR code](/images/integrations/gentex_homelink/homelink-qr-code.png)

- [HomeLink - Apple App Store](https://apps.apple.com/us/app/homelink/id1309231139)

> - [HomeLink - Google Play Store](https://play.google.com/store/apps/details?id=com.gentex.homelink.connect&hl=en-US&pli=1)

Reviewer: ```suggestion
```
I think installation instructions for the app are out of scope for the Home Assistant documentation.

Author: Understood, removed.

## Prerequisites

HomeLink is only compatible with HomeLink devices accessible via the HomeLink App. To prepare for an integration with Home Assistant:

1. Install the HomeLink app through an app store of your choice.
2. Create an account or sign in.
3. Connect to your HomeLink mirror.

> 

Reviewer: ```suggestion

{% include integrations/config_flow.md %}

```
We have this [predefined text block for the configuration section](https://developers.home-assistant.io/docs/documenting/integration-docs-examples#configuration).

Author: Changed

## Configuration

> Upon enabling the HomeLink integration, you will be prompted to enter the email and password for your HomeLink account.

Reviewer: ```suggestion
{% configuration_basic %}
Email:
    description: "The email you used to log in to your HomeLink account."
Password:
    description: "The password to your HomeLink account."
{% endconfiguration_basic %}
```
You can use the [configration_basic block](https://developers.home-assistant.io/docs/documenting/integration-docs-examples#configuration_basic-block) to describe the information entered during setup. Please rephrase if I got the text wrong in the suggestion.

Author: Updated to use the configuration basic block

When you have configured the HomeLink service, you will see your connected HomeLink devices and buttons you have configured on the home screen. These buttons can be connected to events for automations to control lights, music, and more in Home Assistant.

> For more information on how to configure automation and scenes, please see Home Assistant's documentation [here](https://www.home-assistant.io/getting-started/automation/).

Reviewer: ```suggestion
```

Author: Removed

---
title: HomeLink
description: Instructions on how to setup your HomeLink enabled car with Home Assistant.
ha_category:
  - Binary sensor
ha_release: 2025.1
ha_iot_class: Cloud Polling
ha_config_flow: true
ha_codeowners:
  - '@niaexa'
  - '@ryanjones-gentex'
ha_domain: gentex_homelink
ha_platforms:
  - binary_sensor
ha_integration_type: integration
ha_quality_scale: bronze
---

[HomeLink by Gentex](https://www.gentex.com/products-technology/automotive/connected-car/) is the most widely used and trusted car-to-home automation system, with nearly 100 million HomeLink-equipped vehicles on the road today.

HomeLink consists of vehicle-integrated programmable buttons that use radio frequency to conveniently and reliably operate your garage door opener, security gate and other radio frequency-compatible devices.

This integration provides the following platform interfaces:

- Binary sensors: the system detects when the programmable buttons are pressed. Users can use these presses to trigger home automation routines.

> ## The HomeLink® App

Reviewer: Please restructure the HomeLink App section and the Configuration section into a list of prerequisites, and a configuration section. 
you can find an example documentation structure on the [examples page], and in the [developer documentation](https://developers.home-assistant.io/docs/documenting/integration-docs-examples)

For the configuration section, you can use this [standard config_flow text block here](https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L51). See also https://developers.home-assistant.io/docs/documenting/integration-docs-examples#configuration

![HomeLink® app QR code](/images/integrations/gentex_homelink/homelink-qr-code.png)

- [HomeLink® - Apple App Store](https://apps.apple.com/us/app/homelink/id1309231139)
- [HomeLink® - Google Play Store](https://play.google.com/store/apps/details?id=com.gentex.homelink.connect&hl=en-US&pli=1)

## Configuration

After downloading the HomeLink® app for the platform of your choice, creating an account, and pairing your mirror by following the in-app instructions, you may enable the HomeLink integration in Home Assistant via **Settings** > **Devices & Services**.

Log in with the same username and password that you used to sign up on the HomeLink® app.

## Automation and scenes

When you have configured the HomeLink service, you will see the device and buttons you have configured on the home screen.

HomeLink configures a Binary Sensor for each button in your vehicle, typically Button 1, Button 2 and Button 3. These correspond to the buttons (`.`, `..` and `...`) on your HomeLink Device. For example, if you configure an automation that happens when Button 2 is "on", then when you press the (`..`) button, that automation will trigger

> For more information on how to configure automation and scenes, please see Home Assistant's documentation [here](https://www.home-assistant.io/getting-started/automation/)

Reviewer: ```suggestion
For more information on how to configure automation and scenes, please see Home Assistant's documentation [here](https://www.home-assistant.io/getting-started/automation/)

## Removing the integration

This integration follows standard integration removal.

{% include integrations/remove_device_service.md %}
```

We usually add a section on how to remove the integration at the end of the page. Please add any additional steps, if needed. (Or change the description, if the standard removal process doesn't apply to your integration).
